### DISCUSSION TITLE
Why does this code work in 3.6.6 but not in 4.3.1?

### DISCUSSION THREAD

**Discussion by MarcelBeks** [fabricjs/fabric.js#7083] [Q&A] [answered]
I have a canvas with rectangles on them, and want them to not be able to scale outside of the canvas. It works in the 3.6.6 version but not in the 4.3.1 version.
The issue in 4.3.1 is that when for example you drag the rectangle to the center and scale it all the way to one of the borders, you are unable to scale it to the opposite border. But when you do this in 3.6.6 it works as expected, and you can scale it to the opposite border.

Can anyone maybe explain what the difference is between those versions and what causes the issue?
Here are two examples:
[3.6.6](https://jsfiddle.net/0tqampbe/)
[4.3.1](https://jsfiddle.net/nkzyaq4b/4/)

**Comment by asturur** [comment]
Hi MercelBeks, can you please explain a bit better your issue.
What do you mean with `you are unable to scale it to the opposite border.`?

**Reply by MarcelBeks** [reply]
Hi @asturur , Let me explain a bit better.
What i mean is that when you drag a rectangle to the center of the canvas and scale it all the way to for example the right border, you are unable to scale it to the left border. In the older version this is not the case, but in the newer version the code does not work.

**Reply by asturur** [reply]
How do you scale? using which control? pressing which keyboard button?

**Reply by MarcelBeks** [reply]
I use the controls on the rectangles. All of them. Just as soon as it is in contact with a border, it cant be scaled to the opposite border

**Comment by melchiar** [comment, marked as answer]
@MarcelBeks I posted an answer to your [Stack Overflow question](https://stackoverflow.com/a/67746121/7693185). 

The issue you're having is fixed in v4.5.0. It's related to https://github.com/fabricjs/fabric.js/pull/6650
